src/components/v5/frame/ColonyHome/partials/DashboardHeader/partials/ColonyLinks/useMuteColonyItem.tsx

```diff
@@ -1,36 +1,39 @@
 import { BellRinging, BellSimpleSlash } from '@phosphor-icons/react';
-import { useCallback, useState } from 'react';
+import React, { useCallback, useState } from 'react';
 import { defineMessages } from 'react-intl';
+import { toast } from 'react-toastify';
 
 import { useAppContext } from '~context/AppContext/AppContext.ts';
 import { useColonyContext } from '~context/ColonyContext/ColonyContext.ts';
 import { useNotificationsUserContext } from '~context/Notifications/NotificationsUserContext/NotificationsUserContext.ts';
 import { useUpdateUserNotificationDataMutation } from '~gql';
+import Toast from '~shared/Extensions/Toast/index.ts';
 import { formatText } from '~utils/intl.ts';
 import { type DropdownMenuItem } from '~v5/common/DropdownMenu/types.ts';
 
-const MSG = defineMessages({
-  mute: {
-    id: 'headerDropdown.muteNotifications',
-    defaultMessage: 'Mute notifications',
-  },
-  unmute: {
-    id: 'headerDropdown.unmuteNotifications',
-    defaultMessage: 'Unmute notifications',
-  },
-  muting: {
-    id: 'headerDropdown.mutingNotifications',
-    defaultMessage: 'Muting...',
-  },
-  unmuting: {
-    id: 'headerDropdown.unmutingNotifications',
-    defaultMessage: 'Unmuting...',
-  },
-});
-
 const ITEM_KEY = 'headerDropdown.section3.toggleMute';
 
-export const useMuteColonyItem = (): DropdownMenuItem => {
+const useMuteColonyItem = (): DropdownMenuItem => {
+  const MSG = defineMessages({
+    mute: {
+      id: 'headerDropdown.muteNotifications',
+      defaultMessage: 'Mute notifications',
+    },
+    unmute: {
+      id: 'headerDropdown.unmuteNotifications',
+      defaultMessage: 'Unmute notifications',
+    },
+    toastNotificationsUnmuted: {
+      id: `headerDropdown.toastNotificationsUnmuted`,
+      defaultMessage: 'You will now receive notifications from this colony.',
+    },
+    toastNotificationsMuted: {
+      id: `headerDropdown.toastNotificationsMuted`,
+      defaultMessage:
+        'You will no longer receive notifications from this colony.',
+    },
+  });
```

Comment:
Could we define `MSG` outside of the hook? I guess then it is not needed to pass it to the `useEffect`

Reply:
<img width="1539" alt="Screenshot 2024-10-18 at 14 54 03" src="https://github.com/user-attachments/assets/2d267d56-fe31-4038-9a6f-e0f1510f7498">

That is what I had at first, but eslint was complaining about it, though I must admit i'm not entirely sure why. Moving it into the hook solved, the problem, but I never really liked the solution.

I've looked into it and it seems it's a common problem with the lint rule for fast refresh, so I've just ignored that lint rule for this line and added a comment explaining why.

Reply:
Sorry, I'm wrong here, it does break the fast refresh!

<img width="2135" alt="Screenshot 2024-10-18 at 15 08 45" src="https://github.com/user-attachments/assets/ebffd9aa-d2e2-4df5-9047-da5a808e1db0">

The only solution to not break the fast refresh is to declare this inside the hook. I think it's because it is a hook and not a component, and that fact + the way defineMessages works is combining somehow weirdly with the internals of how hot module reloading / fast refresh is working.

Either way, I've moved it back into the hook :)